<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1009</width>
    <height>832</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3D Object Detection Model Training</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_main">
    <item>
     <widget class="QGroupBox" name="trainingGroupBox">
      <property name="title">
       <string>Model Training Module</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_training">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_backbone">
         <item>
          <widget class="QLabel" name="backboneLabel">
           <property name="text">
            <string>Backbone Architecture:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="backboneComboBox">
           <item>
            <property name="text">
             <string>PointPillar</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Point-RCNN</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>PV-RCNN</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_openConfig">
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="openConfigButton">
           <property name="text">
            <string>Open Config File</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_lr">
         <item>
          <widget class="QLabel" name="learningRateLabel">
           <property name="text">
            <string>Learning Rate:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="learningRateLineEdit">
           <property name="text">
            <string>0.01</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_batchSize">
         <item>
          <widget class="QLabel" name="batchSizeLabel">
           <property name="text">
            <string>Batch Size:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="batchSizeSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>32</number>
           </property>
           <property name="value">
            <number>4</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_epoch">
         <item>
          <widget class="QLabel" name="epochLabel">
           <property name="text">
            <string>Epoch:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="epochSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>500</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_pointCloudMagnification">
         <item>
          <widget class="QLabel" name="pointCloudMagnificationLabel">
           <property name="text">
            <string>Point Cloud Magnification Factor:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="pointCloudMagnificationLineEdit">
           <property name="text">
            <string>20</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_labelMagnification">
         <item>
          <widget class="QLabel" name="labelMagnificationLabel">
           <property name="text">
            <string>Label Magnification Factor:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="labelMagnificationLineEdit">
           <property name="text">
            <string>0.85</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="dataDirectoryGroupBox">
      <property name="title">
       <string>Data Directory</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_dataDirectory">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_trainingData">
         <item>
          <widget class="QLabel" name="trainingDataLabel">
           <property name="text">
            <string>Training Data Directory:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="trainingDataLineEdit"/>
         </item>
         <item>
          <widget class="QPushButton" name="browseTrainingDataButton">
           <property name="text">
            <string>Browse</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_labelData">
         <item>
          <widget class="QLabel" name="labelDataLabel">
           <property name="text">
            <string>Label Data Directory:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="labelDataLineEdit"/>
         </item>
         <item>
          <widget class="QPushButton" name="browseLabelDataButton">
           <property name="text">
            <string>Browse</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="includeAugmentedDataCheckbox">
         <property name="text">
          <string>Include augmented data in val/test set</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_buttons">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_buttons">
        <item>
         <spacer name="horizontalSpacer_1">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="startTrainingButton">
          <property name="text">
           <string>Start Training</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="cancelButton">
          <property name="text">
           <string>Cancel</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="resetButton">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1009</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
